### PR TITLE
Creating V2 of the parser

### LINKED ISSUE #128: Unable to convert pipeline
Pasting in azure devops pipeline YAML to https://pipelinestoactions.azurewebsites.net/ and clicking "process YAML" fails to produce any output or error message

1. Go to https://pipelinestoactions.azurewebsites.net/ 
2. paste pipeline YAML in to  'Azure DevOps Pipeline YAML'
3. Click 'process YAML'
4. pipeline YAML is reformatted but no output appears in the 'GitHub Actions YAML' box and no error message 

**Expected behavior**
GitHub Actions YAML box populated with something

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser: Chrome 84.0.4147.89, Firefox 79.0

**Screenshots**
pipelineConvert

Example YAML
```
trigger:
- master

pool: 'Pipeline-Demo-Windows'


variables:
  solution: '**/*.sln'
  buildPlatform: 'Any CPU'
  buildConfiguration: 'Release'

stages:

- stage: Build
  jobs: 
    - job: BuildSpark
      pool:
        name: 'Pipeline-Demo-Windows'
        demands:
        - Agent.OS -equals Windows_NT
      steps:
      - task: NuGetToolInstaller@1

      - task: NuGetCommand@2
        inputs:
          restoreSolution: '$(solution)'

      - task: VSBuild@1
        inputs:
          solution: '$(solution)'
          msbuildArgs: '/p:DeployOnBuild=true /p:WebPublishMethod=Package /p:PackageAsSingleFile=true /p:SkipInvalidConfigurations=true /p:PackageLocation="$(build.artifactStagingDirectory)"'
          platform: '$(buildPlatform)'
          configuration: '$(buildConfiguration)'

      - task: PublishPipelineArtifact@1
        inputs:
          targetPath: '$(build.artifactStagingDirectory)'
          artifact: 'WebDeploy'
          publishLocation: 'pipeline'

- stage: Deploy
  jobs:
    - deployment: 
      variables:
        - name: Art
          value: "Server=.;Database=Art;Trusted_Connection=True;"
        
      environment: 
        name: windows-server
        resourceType: VirtualMachine
        tags: web
      strategy:
        runOnce:
          deploy:
            steps:
              - task: DownloadPipelineArtifact@2
                inputs:
                  buildType: 'current'
                  artifactName: 'WebDeploy'
                  targetPath: '$(Pipeline.Workspace)'


              - task: CmdLine@2
                inputs:
                  script: |
                    echo Write your commands here
                    
                    DIR
                  workingDirectory: '$(Pipeline.Workspace)'
                  
              - task: IISWebAppManagementOnMachineGroup@0
                inputs:
                  IISDeploymentType: 'IISWebsite'
                  ActionIISWebsite: 'CreateOrUpdateWebsite'
                  WebsiteName: 'Spark'
                  WebsitePhysicalPath: '%SystemDrive%\inetpub\wwwroot'
                  WebsitePhysicalPathAuth: 'WebsiteUserPassThrough'
                  AddBinding: true
                  CreateOrUpdateAppPoolForWebsite: true
                  ConfigureAuthenticationForWebsite: true
                  AppPoolNameForWebsite: 'Spark'
                  DotNetVersionForWebsite: 'v4.0'
                  PipeLineModeForWebsite: 'Integrated'
                  AppPoolIdentityForWebsite: 'ApplicationPoolIdentity'
                  AnonymousAuthenticationForWebsite: true
                  WindowsAuthenticationForWebsite: false
                  protocol: 'http' 
                  iPAddress: 'All Unassigned'
                  port: '80'
                  
              - task: IISWebAppDeploymentOnMachineGroup@0
                inputs:
                  WebSiteName: 'Spark'
                  Package: '$(Pipeline.Workspace)\Art.Web.zip'
                  XmlVariableSubstitution: true
```
          
    `

### LINKED ISSUE #151: name/value variable with group in job crashes - there are too many simple/complex combinations
Multiple issues have exposed the weaknesses of the current process. It's time to pivot and find a better way to extract individual jobs out of Azure Pipelines, normalize or process them, and then stitch the GitHub action back together. This is a fairly major change to the underlying processing, but should appear minor to the end user (i.e. the API shouldn't change and this should be a minor version increment).

There are multiple parts to this issue:

1. [x] Similar to [steps](https://github.com/samsmithnz/AzurePipelinesToGitHubActionsConverter/blob/764a63b449a067c7c745a80d883f684397dadc79/src/AzurePipelinesToGitHubActionsConverter.Core/Conversion.cs#L176), create an individual processing for jobs. #153 
    1. [x] ensure that job variations can be handled (Variables)
2. [x] Rework the entire processing to work through three steps #154 
    1. [x] extract the structure
    2. [x] process stages/jobs
    3. [x] put everything back together again
3. [x] Similar to step 1, but for stages, ensuring we have all of the edge cases handled.

### LINKED ISSUE #155: Add support for AzureWebApp@1
Samples:
```
          - task: AzureWebApp@1
            displayName: 'Azure Web App Deploy: {{ webAppName }}'
            inputs:
              azureSubscription: $(azureSubscription)
              appType: webAppLinux
              appName: $(webAppName)
              package: '$(Pipeline.Workspace)/drop/**/target/*.?(war|jar)'
```

```
          - task: AzureWebApp@1
            displayName: 'Azure Web App Deploy: {{ webAppName }}'
            inputs:
              azureSubscription: $(azureSubscription)
              appType: webAppLinux
              appName: $(webAppName)
              runtimeStack: 'NODE|10.10'
              package: $(Pipeline.Workspace)/drop/$(Build.BuildId).zip
              startUpCommand: 'npm run start'
```

```
          - task: AzureWebApp@1
            displayName: 'Deploy Azure Web App : {{ webAppName }}'
            inputs:
              azureSubscription: $(azureSubscription)
              appName: $(webAppName)
              package: $(Pipeline.Workspace)/drop/$(Build.BuildId).zip
```

### PR DESCRIPTION
**Pull request by samsmithnz**
2. [x] Rework the entire processing to work through three steps
    1. [x] extract the structure
    2. [x] process stages/jobs
    3. [x] put everything back together again
    4. [x] Ensure all tests pass in V2 - including the 6 new ones that don't pass in V1
    5. [x] Clean up: review todo's, removed unused usings, and commented out code

fixes #151 
fixes #155 
fixes #128 (before this was fixed with a workaround, now it works without editing/workarounds)

### CODE REVIEW

File: src/AzurePipelinesToGitHubActionsConverter.Tests/CompletePipelineTests.cs [hunk 1]
```diff
@@ -1105,9 +1121,8 @@ public void JRPipelineTest()
   jobs:
     - deployment: 
       variables:
-        Art: ""Server=.;Database=Art;Trusted_Connection=True;""
-        #- name: Art
-        #  value: ""Server=.;Database=Art;Trusted_Connection=True;""
+        - name: Art
```
Comment (samsmithnz) [inline src/AzurePipelinesToGitHubActionsConverter.Tests/CompletePipelineTests.cs:1124]: We can now support the alternative variable syntax everywhere

File: src/AzurePipelinesToGitHubActionsConverter.Tests/CompletePipelineTests.cs [hunk 2]
```diff
@@ -1446,8 +1460,8 @@ public void SSDeploymentPipelineTest()
   ASPNETCOREEnvironmentSetting: Development
   captureStartErrors: true
   cognitiveServicesSubscriptionKey: ${{ env.cognitiveServicesSubscriptionKey }}
-  environment2: ${{ env.prUC }}
-  environment2Lowercase: ${{ env.prLC }}
+  environment: ${{ env.prUC }}
```
Comment (samsmithnz) [inline src/AzurePipelinesToGitHubActionsConverter.Tests/CompletePipelineTests.cs:1463]: We don't need to rename reserved variable names - "environment" no longer clashes with environment on the stage

File: src/AzurePipelinesToGitHubActionsConverter.Tests/StagesTests.cs [hunk 3]
```diff
@@ -288,7 +291,10 @@ public void VariablesWithStageAndConditionalStatementsVariationTest()
 stages:
 - stage: Deploy
   variables:
-    prId: '00B'
+    ${{ if ne(variables['Build.SourceBranchName'], 'master') }}:
```
Comment (samsmithnz) [inline src/AzurePipelinesToGitHubActionsConverter.Tests/StagesTests.cs:294]: Conditional inserts are successfully stripped out now

File: src/AzurePipelinesToGitHubActionsConverter.Tests/TriggerTests.cs [hunk 4]
```diff
@@ -12,57 +11,81 @@ public class TriggerTests
...
+
+        //This test doesn't work with V1
         [TestMethod]
-        public void TriggerNoneSimpleStringTest()
+        public void TriggerAndPRNoneSimpleStringTest()
         {
             //Arrange
-            string input = "trigger: none";
+            string input = @"
+trigger: none
```
Comment (samsmithnz) [inline src/AzurePipelinesToGitHubActionsConverter.Tests/TriggerTests.cs:41]: Edge cases for trigger and pr now covered.
